### PR TITLE
Added method createBuffer

### LINKED ISSUE #5: Looks-same to return the diff as a base64 string
## Need

```
As a developer
I need to have a pluggable application 
So that I can build and manage other application on top of it
```

It would be nice to have a method that returns the `diff` as  a `base64 String`, so the developer can manage it how he wants, to store it into a database, to send it to a server etc.

## Deliverables

* implement the method
* add tests

## Solution

`add a method` to the the `prototype` of `PNGOut`:

```js
  /**
  * @param {base64Cb} callback
  */
  createBuffer: function(callback) {
    this._png.pack(); // behaves like a stream
    
    // collect the data on 'data' events
    // on end event call the callback
    callback(null, buffer);
  }

  /**
  * @callback base64Cb
  * @param error
  * @param {Buffer} buffer
  */
```

adding another option to the options passed to createDiff function:

```js
  /**
  * @param {Object} options
  * @param {Boolean} [options.save]
  */
  createDiff(options, callback) {}
```

the callback passed to `buildDiffImage`, may look like:

```js
function(result) {
  if (opts.save) {
    result.save(opts.path, callback);
  }
  else {
   result.createBase64(callback);
  }
}

### PR DESCRIPTION
**Pull request by flore77**
I have added the `createBuffer` method that we have spoke about. please take a look @SevInf. thanks

fixes #5

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -199,7 +199,12 @@ exports.createDiff = function saveDiff(opts, callback) {
             };
 
         buildDiffImage(result.first, result.second, diffOptions, function(result) {
-            result.save(opts.diff, callback);
+            if (opts.save === false) {
+                result.createBuffer(callback);
+            }
+            else {
```
Comment (LinusU) [inline index.js]: The current coding style is to keep this on the same line, that is: `} else {`

**Comment by LinusU** [comment]
You could use [concat-stream](https://www.npmjs.com/package/concat-stream) instead of doing it yourself in the `createBuffer` method.

How about dropping the `save: false` and let it return a buffer when no `diff: path` is specified?

**Comment by SevInf** [comment]
Can you please squash the commits so it'll be either 1 commit, or 1 commit for code and 1 commit for tests?

>How about dropping the save: false and let it return a buffer when no diff: path is specified?

@LinusU I like this idea.

**Comment by flore77** [comment]
> Can you please squash the commits

Yes, of course, I will squash the commits: 1 for the code and 1 for the tests.


> dropping the save: false 

 I also like the idea to renounce to the `save` option

> You could use concat-stream

I have thought that it is a simple function and I could write it myself, without adding another dependency. But I will use `concat-stream` module.

I will do the changes. thank you both for your review @LinusU  @SevInf

**Comment by LinusU** [comment]
I understand that feeling about modules, I can absolutely recommend reading [Sindre Sorhus AMA answer on the topic](https://github.com/sindresorhus/ama/issues/10#issuecomment-117766328), it's really good.

It's kind of funny actually because I used to write that exact snippet in many modules :smiley: made it shorter and shorter by using `png.on('data', chunks.push.bind(chunks))` and such. But the reality is that it's very cheap to include another module and there will always be an edge case that someone else can fix; that fix will then propagate back to you.

Looks very good otherwise :+1:

**Comment by flore77** [comment]
@SevInf Finally, I decided to rebase all commits in one.
@LinusU Nice article :+1: And you are right, that a fix will propagate back to you, this is the beautiful of open source, that anyone can fix a bug and all dependents will benefit.

please review :smiley:

**Comment by LinusU** [comment]
Looks very good :+1:

**Comment by SevInf** [comment]
Thanks, looks great. Will publish a version in a minute.

**Comment by SevInf** [comment]
Published as 2.1.0. Thanks @flore77 for the code and @LinusU for the help with review.

**Comment by LinusU** [comment]
Thanks @SevInf for following semver to the letter, that's whats making the node ecosystem works! Great work with the module :+1:

**Pull request merged**